.github/workflows/ci.yml

```diff
@@ -359,204 +359,10 @@ jobs:
-              -p mc-fog-ledger-server \
-              -p mc-fog-report-server \
-              -p mc-fog-sql-recovery-db \
-              -p mc-fog-test-client \
-              -p mc-fog-view-server \
-              -p mc-ledger-distribution \
-              -p mc-mobilecoind \
-              -p mc-mobilecoind-dev-faucet \
-              -p mc-util-generate-sample-ledger \
-              -p mc-util-grpc-admin-tool \
-              -p mc-util-keyfile \
-              -p mc-util-seeded-ed25519-key-gen \
-              --release
-
-          BIN_DIR="$PWD/target/release"
-
-          # Run in temp dir to appease check-dirty-git.
-          mkdir -p /tmp/fog-local-network
-          cd /tmp/fog-local-network
-
-          # Generate sample keys and ledger.
-          FOG_AUTHORITY_ROOT=$("$BIN_DIR/mc-crypto-x509-test-vectors" --type=chain --test-name=ok_rsa_head)
-          "$BIN_DIR/sample-keys" --num 10 --seed=AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
-          "$BIN_DIR/generate-sample-ledger" --txs 100
-
-          # Generate sample Fog keys.
-          "$BIN_DIR/sample-keys" --num 4 --output-dir fog_keys --fog-report-url 'insecure-fog://localhost:6200' --fog-authority-root "$FOG_AUTHORITY_ROOT"
-
-      - name: Run local network
-        env:
-          PGHOST: postgres
-          PGUSER: postgres
-        run: |
-          BIN_DIR="$PWD/target/release"
-          SCRIPT_DIR="$PWD/tools/fog-local-network"
-          export MC_CHAIN_ID="local"
-          export MC_LOG=info
-          # This is needed since we want to capture the output of mc-consensus-tool, and we can't have the
-          # logs getting in the way.
-          export MC_LOG_STDERR=1
-
-          cd /tmp/fog-local-network
-          export LEDGER_BASE="$PWD/ledger"
-
-          # Run local network in background.
-          MC_LOG="info,rustls=warn,hyper=warn,tokio_reactor=warn,mio=warn,want=warn,rusoto_core=error,h2=error,reqwest=error,rocket=error,<unknown>=error" \
-          python3 "$SCRIPT_DIR/fog_local_network.py" --network-type dense5 --skip-build &
-
-          # Give it time to spin up
-          for PORT in 3200 3201 3202 3203 3204 4444; do
-            for _unused in $(seq 0 60); do
-              if ss -l | grep -q ":$PORT"; then break; else sleep 1; fi;
-            done
-          done
-
-          # Save some typing
-          export MC_PEER="insecure-mc://localhost:3200/,insecure-mc://localhost:3201/,insecure-mc://localhost:3202/,insecure-mc://localhost:3203/,insecure-mc://localhost:3204/"
-
-          # Run fog-distribution client to exercise Fog
-          echo "Running fog distro"
-          "$BIN_DIR/fog-distribution" \
-              --sample-data-dir . \
-              --max-threads 1 \
-              --num-tx-to-send 10
-
-          # Give it time to quiet down
-          "$BIN_DIR/mc-consensus-tool" wait-for-quiet
-
-          # Run test-client
-          echo "Running test client"
-          "$BIN_DIR/test_client" \
```

Comment:
The test client is used in CD integration tests. My understanding is it's the hearbeat used in mainnet and testnet

Reply:
fog-test-client was originally just used to do a suite of integration tests. It still does that, but also has been modified to run in a "Canary" mode for doing end-to-end monitoring for test/main.